### PR TITLE
Make to compile with new Swift 3 API

### PR DESCRIPTION
**Pull request by rothomp3**
~~There is one caveat, in Sources/Jay/Formatter.swift in `formatObject(object:)`, there seems to be an upstream issue (or something I don't understand), because I could not eliminate an "ambiguous use of member `sort`" on the line `let pairs = object.sort(isOrderedBefore: { (a, b) -> Bool in a.0 <= b.0 })`. Even adding the `isOrderedBefore` label didn't let it compile. So at the moment, the sorting is just commented out.~~ Fixed in a later commit.

### CODE REVIEW

**Comment by czechboy0** [comment]
Thanks for this! I'll take a look at the sorting issue.

What Swift version is this compatible with? I'm assuming it's a build from master, right? Can you let me know exactly which commit you used?

Just so you understand my thinking - I'm considering holding this PR until there's a Swift dev snapshot which already includes the new API guidelines changes & up Jay to 0.4.0, so that people can still use the 0.3.x versions which support the March 1st snapshot of Swift 3.0.

**Comment by czechboy0** [comment]
Oh, also CI will obviously fail here, because it's trying to build Jay with the Swift version specified in `.swift-version`, which is March 1st Swift 3.0-dev at the moment. As I mentioned above, it'd be great to wait for a new snapshot, for you to up the version in `.swift-version` then and thus get a passing CI build.

Is that cool with you?

**Comment by rothomp3** [comment]
Sure, hopefully they'll be coming out with a new snapshot soon. This is the version from my build off of master: `Welcome to Swift version 3.0-dev (LLVM 699a786c15, Clang cf057dcb80, Swift c32fec865a)`

**Comment by rothomp3** [comment]
Ok, this should hopefully pass CI now, even with the swift-build library issue, thanks to adding some environment variables to .travis.yml. I also fixed the sort issue, and the test suite.

**Comment by rothomp3** [comment]
I gave it my best shot. It works on Linux, but not OS X (the .travis.yml thing to tell the dynamic linker where to find the library that swift-build can't find).

**Comment by czechboy0** [comment]
Thanks so much for your great work on this, @rothomp3! Unfortunately it seems like the latest snapshot broke SwiftPM: https://bugs.swift.org/browse/SR-975
I'm split about how to proceed now. I would prefer to keep this in a PR until we get a stable (and a working) snapshot with SwiftPM, so that we can update everything. Having the latest version that needs workarounds it not ideal.

On the other hand - are you able to use this branch in your project? If so, I hope you don't mind using it like this for a bit longer. If not, let's talk about creating a new tag for just this version in a way that doesn't break it for everyone else.

**Comment by rothomp3** [comment]
At the moment, I actually have just been helping out out of the sheer goodness of my heart haha, not actively working on any project that uses this other than vaguely following Vapor development. So leaving it as-is until the next snapshot is fine by me.

**Comment by czechboy0** [comment]
Ah, that's so nice, thank you @rothomp3 :blue_heart: Hopefully we'll be able to merge this right after the next snapshot is released.

**Comment by czechboy0** [comment]
Can you please update `.swift-version` to `DEVELOPMENT-SNAPSHOT-2016-03-24-a`? That should fix CI and then I'll merge :)

**Comment by rothomp3** [comment]
Indeed, it seems to have fixed the CI and everything appears in order.

File: Tests/Jay/PerformanceTests.swift [hunk 1]
```diff
@@ -27,10 +27,14 @@ class PerformanceTests: XCTestCase {
     func urlForFixture(name: String) -> NSURL {
         let url: NSURL
         //if we're running from CLI, use SPM_INSTALL_PATH to find fixtures, otherwise bundle
-        if let path = NSProcessInfo().environment["SPM_INSTALL_PATH"] {
-            url = NSURL(string: "file://\(path)/../Tests/Jay/Fixtures/\(name).json")!
```
Comment (czechboy0) [inline Tests/Jay/PerformanceTests.swift:31]: Can you try to use `#path` as the path to this file to find the fixture path (parent folder...)? I'm not a big fan of using the system call to C to figure it out like this. Sorry I haven't noticed this in my earlier reviews. :)

Comment (czechboy0) [inline Tests/Jay/PerformanceTests.swift:31]: Actually don't worry about it. I'll fix it after merging.

**Comment by czechboy0** [comment]
Thank you so much for this PR, @rothomp3! :tada:

**Pull request merged**